### ISSUE TITLE
Testing/using in conjunction with System Update

### ISSUE THREAD

**Issue by briangig** [jantari/LSUClient#12]
Hi, I am trying to see if I can use LSUClient for auditing purposes, to monitor updates handled by TVSU.

Running on a test machine, TVSU gives me one each of recommended, critical, and optional updates.  If I run `Get-LSUpdate`, it gives me the recommended update (but listed as critical) and the optional update but listed as recommended.  The actual critical update from TVSU is not listed (BIOS update).

Are these two methods not using the same categories when reading updates?  Is LSUClient aware of what TVSU has already installed?

**Comment by jantari** [comment]
Hi, I know what's causing the severities to be different. It's a missing feature in LSUClient for now.

The BIOS Update missing is more important to me, please tell me: 

- what model of computer are you using
- what's your current bios version
- what's the version of the bios offered by tvsu

**Comment by briangig** [comment]
Hi, great info, thanks!

This is a Thinkpad T470s.  I am being offered BIOS version 1.36 by TVSU, I am currently on BIOS version 1.35.



```
PS C:\Windows\system32> $updates = Get-LSUpdate

ID       Title                                                                    Category                   Version Severity    IsApplicable
--       -----                                                                    --------                   ------- --------    ------------
r0yvu28e ThinkPad Hotkey Features Integration Package Setup - 10 [64]             Software and Utilities     9.1.0.3 Recommended True
n1wth09w Thunderbolt Firmware Update (Windows 10 Version 1709 or Later) - 10 [64] Display and Video Graphics 0.0.0.0 Critical    True
```


If search for BIOS updates only, I don't get anything from LSUClient unless I specify -All:

`Get-LSUpdate | Where-Object { $_.Category -eq 'BIOS UEFI' }`

No Results

```
PS C:\Windows\system32> Get-LSUpdate -All | Where-Object { $_.Category -eq 'BIOS UEFI' }

ID       Title                                        Category  Version Severity    IsApplicable
--       -----                                        --------  ------- --------    ------------
n1wuj29w ThinkPad BIOS Update - 7 [32,64]/8.1/10 [64] BIOS UEFI 1.36    Recommended False
```


Is it because the BIOS update isn't "needed"?

**Comment by jantari** [comment]
Well, the BIOS update is "needed" as evidenced by System Update offering it to you - but LSUClient wongfully thinks it's not needed by your computer.

The reason for that is that the BIOS-Update specifies the following in its XML:
```xml
<Dependencies>
  <And>
    <_EmbeddedControllerVersion>
      <Version>^1.20</Version>
    </_EmbeddedControllerVersion>

[...]
```

This means that it depends on, aka needs a specific EC version. I have always interpreted version numbers with `^` in the beginning as "less than", so in this case "must be below 1.20". Your computer has EC Version 1.35 - that's why LSUClient doesn't show you the BIOS update, it considers it not applicable to your system and sets `.IsApplicable` to `$false`.

I am not sure what `^1.20` really means now, and I need to understand how the official System Update tool interprets it to fix the behavior of LSUClient.

If you want to take a look at the XML for that BIOS update yourself, it's this:
https://download.lenovo.com/pccbbs/mobiles/n1wuj29w_2_.xml

**Comment by jantari** [comment]
The BIOS Update missing should be fixed by 15feb64 - this was the same problem as issue #14

**Comment by jantari** [comment]
The fix for this is released with version 1.1.2 🎉
